.palantir/revapi.yml

```diff
@@ -1058,6 +1058,11 @@ acceptedBreaks:
       new: "method void org.apache.iceberg.encryption.PlaintextEncryptionManager::<init>()"
       justification: "Deprecations for 1.6.0 release"
   "1.6.0":
+    org.apache.iceberg:iceberg-api:
+    - code: "java.class.defaultSerializationChanged"
+      old: "class org.apache.iceberg.expressions.ResidualEvaluator"
+      new: "class org.apache.iceberg.expressions.ResidualEvaluator"
+      justification: "local testing"
```

Comment:
Seeing how we can get rid of this...

Reply:
If we do what I mentioned below we should be able to get rid of this.

Reply:
Why was it necessary to modify ResidualEvaluator?

Reply:
It's not really needed, https://github.com/apache/iceberg/pull/11180#discussion_r1797258000 we can just set the residual evaluator once we have that info

Reply:
@rdblue Will use the example of the existing `FileScanTaskParser`. Currently when deseriaizing FIleScanTasks in the `fromJson` of this parser (which is similar to the RESTFileScanTaskParser`, in  order to construct a `BaseFileScanTask` it takes in a `ResidualEvaluator` https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/FileScanTaskParser.java#L117

The Residual current constructor currently requires a `spec` as well as a `caseSensitive` flag https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/expressions/ResidualEvaluator.java#L84. At the time of deserialization we do not have either, hence an additional constructor was created which only passes in the `filter`.

Reply:
Following @amogh-jahagirdar suggestion, so will remove this.